src/cascadia/TerminalControl/TermControlContentManagement.cpp

```diff
@@ -0,0 +1,155 @@
+                                  TraceLoggingKeyword(TIL_KEYWORD_TRACE));
+                displayError = false;
+                break;
+
+            case WAIT_TIMEOUT:
+                // This should be impossible.
+                TraceLoggingWrite(g_hTerminalControlProvider,
+                                  "TermControl_ContentWaitTimeout",
+                                  TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
+                                  TraceLoggingKeyword(TIL_KEYWORD_TRACE));
+                break;
+
+            default:
+            {
+                // Returning any other value is invalid. Just die.
+                const auto gle = GetLastError();
+                TraceLoggingWrite(g_hTerminalControlProvider,
+                                  "TermControl_WaitFailed",
+                                  TraceLoggingUInt64(gle, "lastError", "The result of GetLastError"),
+                                  TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
+                                  TraceLoggingKeyword(TIL_KEYWORD_TRACE));
+                break;
+            }
+            }
+        }
+        catch (...)
+        {
+            // Theoretically, if window[1] dies when we're trying to get
+            // its PID we'll get here. We can probably just exit here.
+
+            TraceLoggingWrite(g_hTerminalControlProvider,
+                              "TermControl_ExceptionInWaitThread",
+                              TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
+                              TraceLoggingKeyword(TIL_KEYWORD_TRACE));
+        }
+        return displayError;
+    }
+
+    void TermControl::_createContentWaitThread()
+    {
+        _contentWaitThread = std::thread([weakThis = get_weak(), contentPid = _contentProc.GetPID(), contentWaitInterrupt = _contentWaitInterrupt.get()] {
+            if (s_waitOnContentProcess(contentPid, contentWaitInterrupt))
+            {
+                // When s_waitOnContentProcess returns, if it returned true, we
+                // should display a dialog in our bounds to indicate that we
+                // were closed unexpectedly. If we closed in an expected way,
+                // then s_waitOnContentProcess will return false.
+                if (auto control{ weakThis.get() })
+                {
+                    control->_raiseContentDied();
+                }
+            }
+        });
+    }
+
+    winrt::fire_and_forget TermControl::_raiseContentDied()
+    {
+        auto weakThis{ get_weak() };
+        co_await winrt::resume_foreground(Dispatcher());
+
+        if (auto control{ weakThis.get() })
+        {
+            if (auto loadedUiElement{ FindName(L"ContentDiedNotice") })
+            {
+                if (auto uiElement{ loadedUiElement.try_as<::winrt::Windows::UI::Xaml::UIElement>() })
+                {
+                    uiElement.Visibility(Visibility::Visible);
+                }
+            }
+        }

```

Comment:
You might have to check `_IsClosing()` here, because all asynchronous UI code should check that IIRC. 🤔